File: Baekjoon/Class2/38.1966 프린터 큐.js
const inputFile = process.platform === "linux" ? "/dev/stdin" : "./input.txt";
const input = require("fs").readFileSync(inputFile, "utf8").trim().split("\n");

let cursor = 1;
const runPrinter = (out) => {
    const m = Number(input[cursor++].split(" ")[1]);
    const priorities = input[cursor++].split(" ").map(Number); // 우선순위
    const printerQueue = priorities.map((val, i) => ({idx: i, priority: val})); // {인덱스, 우선순위}
    priorities.sort((a, b) => b - a); // 우선순위 내림차순 정렬
    
    // 우선순위가 높은 것부터 출력
    let order = 0; // 출력 순서
    while(true) {
        const frontDoc = printerQueue.shift();
        if(frontDoc.priority === priorities[order]) { // 해당 우선순위 찾음
            if(frontDoc.idx === m) { // 출력
                out.push(order + 1);
                break;
            }
            order++;
        }
        else // 해당 우선순위 못 찾음
            printerQueue.push(frontDoc) // 회전
    }
}

const testCase = Number(input[0]);
const out = [];

for(let _ = 0; _ < testCase; _++)
    runPrinter(out);
console.log(out.join("\n"));

File: Baekjoon/Class2/38.1966 프린터 큐.py
from collections import deque
import sys
input = sys.stdin.readline
write = sys.stdout.write

def run_printer():
    _, m = map(int, input().split())
    priorities = list(map(int, input().rstrip().split())) #우선순위
    printer_queue = deque(enumerate(priorities)) #(인덱스, 우선순위)
    priorities.sort(reverse=True) #우선순위 내림차순 정렬

    #우선순위가 높은 것부터 출력
    order = 0 #출력 순서
    while True:
        front_idx, front_priority = printer_queue.popleft()
        if front_priority == priorities[order]: #해당 우선순위 찾음
            if front_idx == m: #출력
                write(f"{order + 1}\n")
                break
            order += 1
        else: #해당 우선순위 못 찾음
            printer_queue.append((front_idx, front_priority)) #회전

def main():
    test_case = int(input())
    for _ in range(test_case):
        run_printer()

main()